crates/walrus-service/src/client/daemon/routes.rs

```diff
@@ -351,6 +425,225 @@ impl From<ClientError> for GetBlobError {
+    // Track the first blob's attribute for header population
+    let mut blob_ids = Vec::new();
+    let mut first_attribute: Option<BlobAttribute> = None;
+    for (index, id_str) in id_strings.iter().enumerate() {
+        match parse_and_resolve_id(id_str.as_str(), client.as_ref()).await {
+            Ok((blob_id, attribute)) => {
+                blob_ids.push(blob_id);
+                // Only capture the first blob's attribute
+                if index == 0 {
+                    first_attribute = attribute;
+                }
+            }
+            Err(error) => return error.into_response(),
+        }
+    }
+
+    // Create streaming response
+    let client_clone = client.clone();
+    let stream = stream::iter(blob_ids).then(move |blob_id| {
+        let client = client_clone.clone();
+        async move {
+            match client.read_blob(&blob_id, consistency_check).await {
+                Ok(blob_data) => Ok(Bytes::from(blob_data)),
+                Err(error) => {
+                    // Map ClientError to appropriate ConcatBlobError
+                    let concat_error = match error.kind() {
+                        ClientErrorKind::BlobIdDoesNotExist => ConcatBlobError::BlobNotFound {
+                            blob_id: blob_id.to_string(),
+                        },
+                        ClientErrorKind::BlobIdBlocked(_) => ConcatBlobError::BlobBlocked {
+                            blob_id: blob_id.to_string(),
+                        },
+                        ClientErrorKind::BlobTooLarge(size) => ConcatBlobError::BlobTooLarge {
+                            blob_id: blob_id.to_string(),
+                            size: *size,
+                        },
+                        _ => ConcatBlobError::BlobReadFailed {
+                            blob_id: blob_id.to_string(),
+                            message: error.to_string(),
+                        },
+                    };
+                    Err(concat_error)
+                }
+            }
+        }
+    });
+
+    // Convert stream to body
+    let body = Body::from_stream(stream);
+
+    // Build response with appropriate headers
+    let mut response = Response::builder()
+        .status(StatusCode::OK)
+        .header(CONTENT_TYPE, "application/octet-stream")
+        .header(CACHE_CONTROL, "max-age=86400, stale-while-revalidate=3600")
+        .header(X_CONTENT_TYPE_OPTIONS, "nosniff")
+        .body(body)
+        .expect("failed to build concatenated blob response");
+
+    // Populate response headers with priority:
+    // 1. Request header override (if Content-Type in request)
+    // 2. First blob's attributes (if available)
+    // 3. Default (already set above)
+    if let Some(content_type) = request_headers.get(CONTENT_TYPE) {
+        response
+            .headers_mut()
+            .insert(CONTENT_TYPE, content_type.clone());
+    } else if let Some(attribute) = first_attribute {
```

Comment:
Question: Is it intentional that this behavior is different from other endpoints like the `get_blob_by_object_id`? Also, do we need any standard headers as set by `populate_response_headers_from_request`? 

Remark: Already now the behavior seems to be a bit inconsistent...

Reply:
We may want to deprecate the `Content-type` api since it is non-standard in HTTP.